Proposal: New static, accessible website

lexrexx opened the pull request:
### **Overview**
This PR introduces a complete redesign of the Bitcoin Knots website, migrating content and modernizing the user interface from the `bitcoinknots.org` repository to the official `knots-www` repository.

### **Source Repositories**
- **New Website Source**: https://github.com/lexrexx/bitcoinknots.org
  - Contains the redesigned website with modern HTML/CSS/JS
  - Features improved navigation, responsive design, and updated content
- **Current Target**: https://github.com/bitcoinknots/knots-www (this repository)
- **Fork Used for PR**: https://github.com/lexrexx/knots-www

### **Major Changes Made**

#### **1. Website Architecture & Structure**
- **Key Improvements**:
  - Brand new design with stronger branding.
  - Responsive design for mobile/desktop compatibility
  - Improved accessibility standards

#### **2. Visual Design & User Experience**
- Complete visual overhaul with modern framework
- Enhanced typography and color scheme for better readability
- Intuitive navigation menu structure
- Consistent styling.

#### **3. Content Updates**
- Added About Bitcoin Knots section
- Added FAQs section.
- Updated Bitcoin Knots features & descriptions
- Current release information and download links
- Improved downloads structure
- Clearer verifcation and usage instructions

#### **4. Technical Improvements**
- Optimized asset loading (minification where applicable)
- Better SEO meta tags and structure
- Improved page load performance
- Cross-browser compatibility fixes

#### **5. Navigation & Information Architecture**
- Streamlined main navigation
- Logical content hierarchy
- Clear calls-to-action for downloads
- Footer with essential links and references

### **Migration Strategy**
This PR replaces the **entire website content** of `knots-www` with the redesigned version from `bitcoinknots.org`, maintaining only Git history for the repository itself.

### **Testing Performed**
- [x] Cross-browser compatibility (Chrome, Firefox, Safari)
- [x] No JavaScript compatibility
- [x] Mobile responsiveness testing
- [x] Link validation
- [x] Page load performance

### **Benefits of This Update**
1. **Modern User Experience**: Visitors get a contemporary, professional website
2. **Better Maintainability**: Organized code structure makes future updates easier and faster. [e.g. main [app.config](https://github.com/lexrexx/knots-www/blob/lexrexx-website-update/app/app.config.ts) for global download links, versioning, etc.]
3. **Improved Performance**: Optimized assets and structure
4. **Enhanced Accessibility**: WCAG guidelines consideration
5. **Mobile-First Design**: Works well on all device sizes

### **Note on Implementation**
This is a **complete replacement** rather than incremental changes because:
1. The new website has a fundamentally different design and structure 
2. It's more efficient to migrate the complete working version
3. Ensures consistency and avoids merge conflicts with legacy code

### **Future Maintenance**
- All future website updates should be made to this new structure
- The `bitcoinknots.org` repository can be archived after this migration
- Documentation for content updates is included in the new structure

### **Deployment Instructions**
Once merged, the changes will be immediately reflected on the live website (assuming automatic deployment from `main` branch is configured with gh-pages).

Alternatively, website can be hosted on bitcoinknots.org server by following these steps:

```bash
# Step 1: Update baseUrl by removing/commenting line (9) in nuxt.config.ts
nano nuxt.config.ts

# Step 2: Generate static build
npm run generate

# Step 3 (optional): preview the build using
npx serve .output/public  

# Step 4: Copy build files to your server public directory
cp -r .output/public/* bitcoinknots.org/
```

lexrexx committed:
Update website with new design and content from bitcoinknots.org

- Complete website redesign
- Updated navigation and layout
- New content structure
- Improved user experience

lexrexx committed:
Update baseUrl for knots-www

lexrexx commented:
I've been working on a modernized version of the Bitcoin Knots website in a separate repository - [bitcoinknots.org](https://lexrexx.github.io/bitcoinknots.org/) - and would like to propose migrating it to the official repository. The new design improves usability, maintains all existing content, and provides a better foundation for future updates.


**Key Questions to Address:**
1. Are there any specific files in the current `knots-www` that should be preserved (beyond `.git`)?
2. Is there a specific deployment process I should be aware of?
3. Would you prefer a different approach?

KurtisStirling commented:
This is a great upgrade but NACK from me due to unnecessary brand changes which would make the website inconsistent with every other bitcoin knots asset. This is an easy fix and I've given Lex extensive information for how to rectify this. If the branding is changed to be inline with current branding, this will become a no-brainer merge; great work.

lexrexx committed:
Update: primary brand color

lexrexx commented:
### Update Applied

@KurtisStirling, I have updated the PR.

The primary brand color has been aligned with the OG Bitcoin orange. The core improvements of this PR remain fully intact.

The PR is now updated and ready for final review.
